### PR TITLE
feat(bridge-history): add Puffer gateways

### PR DESCRIPTION
**Pull request by colinlyguo**
### PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [x] feat: A new feature

### Deployment tag versioning

Has `tag` in `common/version.go` been updated or have you added `bump-version` label to this PR?

- [x] Yes

### Breaking change label

Does this PR have the `breaking-change` label?

- [x] No, this PR is not a breaking change

### CODE REVIEW

**Review by georgehao — DISMISSED** [review]

File: bridge-history-api/internal/config/config.go [hunk 1]
```diff
@@ -23,6 +23,7 @@ type FetcherConfig struct {
 	DAIGatewayAddr           string `json:"DAIGatewayAddr"`
 	USDCGatewayAddr          string `json:"USDCGatewayAddr"`
 	LIDOGatewayAddr          string `json:"LIDOGatewayAddr"`
+	PUFFERGatewayAddr        string `json:"PUFFERGatewayAddr"`
```
Comment (Thegaram) [inline bridge-history-api/internal/config/config.go]:
```suggestion
	PufferGatewayAddr        string `json:"PufferGatewayAddr"`
```

File: bridge-history-api/internal/logic/l2_fetcher.go [hunk 2]
```diff
@@ -85,7 +85,12 @@ func NewL2FetcherLogic(cfg *config.FetcherConfig, db *gorm.DB, client *ethclient
 
 	if common.HexToAddress(cfg.LIDOGatewayAddr) != (common.Address{}) {
 		addressList = append(addressList, common.HexToAddress(cfg.LIDOGatewayAddr))
-		gatewayList = append(gatewayList, common.HexToAddress(cfg.USDCGatewayAddr))
+		gatewayList = append(gatewayList, common.HexToAddress(cfg.LIDOGatewayAddr))
```
Comment (Thegaram) [inline bridge-history-api/internal/logic/l2_fetcher.go:88]: This didn't break anything before?

**Review by Thegaram — DISMISSED** [review]
Just a suggestion

File: bridge-history-api/internal/logic/l2_fetcher.go (hunk 2 above)
Comment (colinlyguo) [reply bridge-history-api/internal/logic/l2_fetcher.go:88]: Luckily it didn't, nobody sent a reverted transaction to this gateway yet.
The `gatewayList` is only used to track failed transactions.
https://scrollscan.com/address/0x8aE8f22226B9d789A36AC81474e633f8bE2856c9

**Review by Thegaram — APPROVED** [review]

**Review by zimpha — APPROVED** [review]

**Pull request merged**